### PR TITLE
take into account line joining when setting eval result extmark

### PR DESCRIPTION
**Pull request by philomates**
after https://github.com/Olical/conjure/commit/6b441676a9c67380d6e43857f73d94b47d1291fb I'm seeing the following error in the log buffer when running tests via conjure
```
; --------------------------------------------------------------------------------
; run-ns-tests: integration.foo-test
; 
; Testing integration.foo-test
; --------------------------------------------------------------------------------
; localhost:33823 (/home/mates/.vim/plugged/conjure/lua/conjure/log.lua:367: line value outside range): .nrepl-port ....
; --------------------------------------------------------------------------------
; localhost:33823 (/home/mates/.vim/plugged/conjure/lua/conjure/log.lua:367: line value outside range): .nrepl-port ...
; --------------------------------------------------------------------------------
; localhost:33823 (/home/mates/.vim/plugged/conjure/lua/conjure/log.lua:367: line value outside range): .nrepl-port ...
```

This PR seems to fix it

### CODE REVIEW

File: fnl/conjure/log.fnl [hunk 1]
```diff
@@ -358,7 +356,7 @@
               jump-to-latest? (config.get-in [:log :jump_to_latest :enabled])]
           (nvim.buf_set_extmark
             buf state.jump-to-latest.ns
-            (a.inc old-lines) 0
+            (+ new-line-index old-lines 2) 0
```
Comment (philomates) [inline fnl/conjure/log.fnl:359]: the `2` is a "magic" number for me, without it the evaluation result doesn't show

**Comment by philomates** [comment]
hmm, not sure actually if this is the right thing. I'm having trouble getting the feature working at all. I'll have to play with it a bit more

**Comment by Olical** [comment]
Looking at this now!

**Comment by philomates** [comment]
Thanks!

> Looking at this now!

**Comment by Olical** [comment]
Let me know if the changes on develop fix this! I took my own approach instead of your PR just so I could understand the issue and hopefully fix the root cause while also guarding against future errors.

Worst case now should be that 1% of weird messages that produce line counts I don't expect won't jump to latest now instead of throwing errors.

**Comment by Olical** [comment]
Oh I see! I've been trying to line it up with the first character of the actual output because if you think of this:

```fennel
;; Some thing!
{:foo :bar}
```

I'll _probably_ want to evaluate/edit/yank the `{:foo :bar}` or even jump to the matching brace with `%`. So I thought starting on the first character was usually the most useful thing. If moving to the very first character of the output is best then maybe we can do that instead? I just envision people "jumping to latest" then down a line every time, so I thought starting on that output line made more sense.

**Comment by philomates** [comment]
https://github.com/Olical/conjure/commit/2fc39c5d80acdb14a150ed09e4a0da41bcd77dba seems to fix the print thing, thanks!

> I'll probably want to evaluate/edit/yank the {:foo :bar} or even jump to the matching brace with %

that makes sense. Honestly I didn't really think through the difference of std out stuff and eval results for this feature. Setting the cursor to the top of either makes sense for different reasons. Someone on slack suggested a way to jump forward/backward through eval results, so maybe something like that could help with this also. 

Let's leave it how it is and as I use conjure more in my dev work I'll try to think of a nice way to join all this

**Comment by Olical** [comment]
Awesome! Closing for now and will get this released in a sec, we can always tweak more after using it for a while.

**Pull request closed without merging**